backup/moodle2/backup_opencast_block_task.class.php

```diff
@@ -73,14 +98,70 @@ protected function define_my_settings() {
                     }
 
                     if (count($videostobackup) > 0) {
-                        $setting = new backup_block_opencast_setting('opencast_videos_include_' . $ocinstance->id,
-                            base_setting::IS_BOOLEAN, boolval($defaultimportvalue));
-                        $setting->get_ui()->set_label(get_string('backupopencastvideos', 'block_opencast', $ocinstance->name));
-                        $this->add_setting($setting);
-                        $this->plan->get_setting('blocks')->add_dependency($setting);
-                        break;
+                        // Here we make sure that the main inclusion happens only once.
+                        if (!$ocinstanceisincluded) {
+                            $this->add_setting($setting);
+                            $this->plan->get_setting('blocks')->add_dependency($setting);
+                        }
+                        // Section Level setting for series.
+                        $seriessettingname = 'opencast_videos_series_' . $seriesobj->identifier . '_included';
```

Comment:
series and episodes do not need the instance id, as it is included in `$includesettingname`, which works as a wrapper section in a top-level. And at the end there would a single xml file for each instance separately!

Have a look at:
- line ~75 => `$includesettingname`
- Ultimately, line ~213 => the single xml file name for each ocinstance

Reply:
The problem is that while building the backup/restore plan, the task identifies its settings (for example in `get_setting()`) only by name, not by the hierarchy, and it even [throws an error there](https://github.com/moodle/moodle/blob/main/backup/util/plan/base_task.class.php#L92) if it finds two settings with the same name.

I think it is unlikely that two different instances will have the same series identifier, but in case someone duplicates their opencast instance or something like that, we probably want to tread safer by including the instanceid in the settingname, just to guarantee the uniqueness of the settingname